docs/template_plugin/tests/functional/op_reference/experimental_detectron_roi_feature_extractor.cpp

```diff
@@ -0,0 +1,236 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <ie_core.hpp>
+#include <ie_ngraph_utils.hpp>
+#include <ngraph/ngraph.hpp>
+#include <shared_test_classes/base/layer_test_utils.hpp>
+#include <tuple>
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+using namespace InferenceEngine;
+using namespace reference_tests;
+
+struct ExperimentalROIFunctional {
+    std::shared_ptr<Function> create_function(const std::vector<Tensor>& inputs) {
+        op::v6::ExperimentalDetectronROIFeatureExtractor::Attributes attrs;
+        attrs.aligned = false;
+        attrs.output_size = 3;
+        attrs.sampling_ratio = 2;
+        attrs.pyramid_scales = {4};
+
+        const size_t num_of_inputs = inputs.size();
+        NodeVector node_vector(num_of_inputs);
+        ParameterVector parameter_vector(num_of_inputs);
+        for (size_t i = 0; i < num_of_inputs; ++i) {
+            const auto& current_input = inputs[i];
+            auto current_parameter = std::make_shared<op::Parameter>(current_input.type, current_input.shape);
+            node_vector[i] = current_parameter;
+            parameter_vector[i] = current_parameter;
+        }
+
+        auto roi = std::make_shared<op::v6::ExperimentalDetectronROIFeatureExtractor>(node_vector, attrs);
+        auto fun = std::make_shared<Function>(OutputVector{roi->output(0), roi->output(1)}, parameter_vector);
+        return fun;
+    }
+};
+
+struct ExperimentalROIParams {
+    ExperimentalROIParams(const std::shared_ptr<ExperimentalROIFunctional>& functional,
```

Comment:
I don't understand why do you need to have shared pointer to some structure which is needed only for one function.

Reply:
Fixed.